config/config.go

```diff
@@ -68,6 +68,37 @@ func New(uri string, requests, concurrency int, requestTimeout, globalTimeout ti
 	}
 }
 
+// Validate returns the config and a not nil ErrInvalid if any of the fields provided by the user is not valid
+func (cfg Config) Validate() (Config, error) {
+	inputErrors := []error{}
+
+	_, err := url.ParseRequestURI(cfg.Request.URL.String())
+	if err != nil {
+		inputErrors = append(inputErrors, fmt.Errorf("-url: %s is not a valid url", cfg.Request.URL.String()))
+	}
+
+	if cfg.RunnerOptions.Requests < 0 {
```

Comment:
Can we use the new rules for numbers now (`n == -1 || n > 0`)?

Note: if we do, we'll need to update `semimpl.Do`, so we probably should do the error handling for it in the mean time

Reply:
Let's just do it now!